.github/workflows/default.yaml

```diff
@@ -0,0 +1,131 @@
+  # tries to balance the two concerns.
+  CHECKOUT_FETCH_DEPTH: 1001
+
+jobs:
+
+  functionality-tests:
+
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Install prerequisite packages
+        run: |
+          sudo apt-get --quiet=2 update
+          sudo apt-get --quiet=2 install libtool-bin
+
+      - name: Setup a nodejs environment
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: Checkout Squid sources
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: ${{ env.CHECKOUT_FETCH_DEPTH }}
+
+      - run: ./bootstrap.sh
+      - run: ./configure --with-openssl
+      - run: make -j2
+      - run: |
+          sudo make install
+          sudo chown -R nobody:nogroup /usr/local/squid
+
+      - run: ./test-suite/test-functionality.sh
+
+      # Squid logs are not readable to actions/upload-artifact below
+      - name: Prep test logs
+        if: success() || failure()
+        run: sudo chmod -R a+rX /usr/local/squid
+
+      - name: Publish test logs
+        if: success() || failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: test-logs
+          path: |
+            ${{ runner.temp }}/*.log
+            /usr/local/squid/var/logs/overlord/*.log
+
+  source-maintenance-tests:
+
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Install prerequisite packages
+        run: |
+          sudo apt-get --quiet=2 update
+          sudo apt-get --quiet=2 install astyle
+          sudo apt-get --quiet=2 install gperf
+          pip install \
+              --user \
+              --no-cache-dir \
+              --disable-pip-version-check \
+              --quiet \
+              --progress-bar off \
+              codespell==1.16 # TODO: Upgrade to codespell v2
+
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: ${{ env.CHECKOUT_FETCH_DEPTH }}
+
+      - run: ./test-suite/test-sources.sh
+
+  build-tests:
+
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ ubuntu-22.04, macos-11 ]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+
+      - name: Install prerequisite Linux packages
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt-get --quiet=2 update
+          sudo apt-get --quiet=2 install libtool-bin
+          sudo apt-get --quiet=2 install libcppunit-dev
+          sudo apt-get --quiet=2 install libldap2-dev
+          # TODO: Add some deb-src URI to sources.list and do build-dep
+          # instead of the above installs?
+          # sudo apt-get --quiet=2 build-dep squid
+
+      - name: Install prerequisite macOS packages
+        if: runner.os == 'macOS'
+        run: |
+           brew install automake
+           brew install cppunit
+
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - run: ./test-builds.sh
+        # XXX: Fix distclean cleanup on macOS instead
+        continue-on-error: ${{ runner.os == 'macOS' }}
```

Comment:
> > Alex: This PR does not make MacOS officially supported.

> My understanding is that the checks performed by this PR will be required to be passed before any PR following it is allowed to merge.

The reality is different: The MacOS test in this PR currently fails (due to bugs already present in the current official sources and preserved in this PR sources). Yet, that failure does not preclude this PR or any other PR from passing all the tests because that failure is explicitly ignored (the poorly named GitHub Actions `continue-on-error` directive has that effect).

> Having MacOS as part of the mandatory tests implicitly adds it to the "officially supported" OS set.

That does not happen in this PR because MacOS build failures are ignored.

> > Alex: We should make that decision separately. MacOS is included in the workflow configuration because it illustrates GitHub Actions matrix testing capabilities. We may decide to officially support it, to completely remove it from the matrix, or even something in-between.
 
> Amos: Then please drop MacOS from this PR before it merges. As a demonstration/test that we can check multiple OS it has served that purpose already. It can be re-added when a formal policy decision is made to support that OS.

Will do. Doing so will re-hide MacOS build failures from any developer that wants to fix them. I fixed a couple while this test was included, but I did not fix all of them, and do not plan on fixing any more, especially since they will become invisible again.

Reply:
> please drop MacOS from this PR before it merges

Done at eb34c13.